In [1]:
import pandas as pd
import numpy as np
import scipy as sp

# Лабораторная работа №2
## Задание 3
## Кондрашин Тимофей 211-365 Вариант 60 (20.04.2023)

In [2]:
data = pd.DataFrame([50.0, 43.7, 35.2, 45.6, 39.8, 37.2, 36.7, 46.2, 35.9, 36.4, 36.7, 48.5, 39.6, 40.6, 45.2, 48.4])
data.T

0     1     2     3     4     5     6     7     8     9     10    11   
0  50.0  43.7  35.2  45.6  39.8  37.2  36.7  46.2  35.9  36.4  36.7  48.5  \

     12    13    14    15  
0  39.6  40.6  45.2  48.4

In [3]:
data.describe()

0
count  16.000000
mean   41.606250
std     5.131922
min    35.200000
25%    36.700000
50%    40.200000
75%    45.750000
max    50.000000

### Всего элементов ряда

In [4]:
data_len = len(data)
data_len

16

### Выборочное среднее

$$\overline x = \frac1n \sum^n_{i=1} x_i$$

In [5]:
mean = np.mean(data)
mean

41.60625

### Выборочная дисперсия

$$s^2_x = \frac{1}{n-1}\sum^n_{i=1} (x_i - \overline x)^2$$

In [6]:
var = np.var(data)
var

0    24.690586
dtype: float64

### Среднеквадратическое отклонение

$$s = \sqrt{s^2}$$

In [7]:
std = np.std(data)
std

0    4.968962
dtype: float64

### $t$-критерий Стьюдента

$$t = \frac{\overline x - \mu}{\frac{s}{\sqrt{n}}}$$

In [8]:
gamma = 0.95
t_value = sp.stats.t.ppf(gamma, data_len-1)
t_value

1.7530503556925547

### Доверительный интервал для математического ожидания при доверительной вероятности $\gamma$

#### Z-оценка

$$z={x-{\bar {X}} \over S_{x}}$$

In [9]:
sp.stats.norm.ppf(.95)

1.6448536269514722

#### Доверительный интервал для математического ожидания через $z$ оценку

$$\left(\overline x - z \cdot \frac{s}{\sqrt{n}}, \overline x + z \cdot \frac{s}{\sqrt{n}}\right)$$

In [10]:
def confidence_interval(data, gamma):
    mean = np.mean(data)
    std = np.std(data)
    n = len(data)
    z = sp.stats.norm.ppf(.95)
    return (mean - z * std / np.sqrt(n), mean + z * std / np.sqrt(n))

confidence_interval(data, 0.95)

(0    39.562946
 dtype: float64,
 0    43.649554
 dtype: float64)

#### Доверительный интервал для математического ожидания через $t$-критерий Стьюдента

$$\left(\overline x - t \cdot \frac{s}{\sqrt{n}}, \overline x + t \cdot \frac{s}{\sqrt{n}}\right)$$

In [11]:
def confidence_interval_t(data, gamma):
    mean = np.mean(data)
    std = np.std(data)
    n = data_len
    t = t_value
    return (mean - t * std / np.sqrt(n), mean + t * std / np.sqrt(n))

confidence_interval_t(data, 0.95)

(0    39.42854
 dtype: float64,
 0    43.78396
 dtype: float64)

#### Вывод

Таким образом с вероятностью 0,95 можно утверждать, что математическое ожидание лежит в интервале (39.4; 43.7)